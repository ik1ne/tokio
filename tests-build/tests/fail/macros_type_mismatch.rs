use tests_build::tokio;

#[tokio::main]
async fn missing_semicolon_or_return_type() {
    Ok(())
}

#[tokio::main]
async fn missing_return_type() {
    return Ok(());
}

#[tokio::main]
async fn extra_semicolon() -> Result<(), ()> {
    Ok(());
}

// https://github.com/tokio-rs/tokio/issues/4635
#[allow(redundant_semicolons)]
#[rustfmt::skip]
#[tokio::main]
async fn issue_4635() {
    return 1;
    ;
}

#[tokio::main]
async fn extra_semicolon_with_try_operator() -> Result<(), ()> {
    Ok(())?;
    Ok(());
}

#[tokio::main]
async fn returns_different_type() -> usize {
    return "hello world";
}

fn main() {}
