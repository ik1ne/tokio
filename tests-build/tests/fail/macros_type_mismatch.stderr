error[E0308]: mismatched types
 --> tests/fail/macros_type_mismatch.rs:5:5
  |
5 |     Ok(())
  |     ^^^^^^ expected `()`, found `Result<(), _>`
  |
  = note: expected unit type `()`
                  found enum `Result<(), _>`

error[E0308]: mismatched types
  --> tests/fail/macros_type_mismatch.rs:10:12
   |
10 |     return Ok(());
   |            ^^^^^^ expected `()`, found `Result<(), _>`
   |
   = note: expected unit type `()`
                   found enum `Result<(), _>`

error[E0308]: mismatched types
  --> tests/fail/macros_type_mismatch.rs:14:46
   |
14 |   async fn extra_semicolon() -> Result<(), ()> {
   |  ______________________________________________^
15 | |     Ok(());
   | |           - help: remove this semicolon to return this value
16 | | }
   | |_^ expected `Result<(), ()>`, found `()`
   |
   = note:   expected enum `Result<(), ()>`
           found unit type `()`

error[E0308]: mismatched types
  --> tests/fail/macros_type_mismatch.rs:23:12
   |
21 | #[tokio::main]
   |               - help: try adding a return type: `-> i32`
22 | async fn issue_4635() {
23 |     return 1;
   |            ^ expected `()`, found integer

error[E0308]: mismatched types
  --> tests/fail/macros_type_mismatch.rs:28:64
   |
28 |   async fn extra_semicolon_with_try_operator() -> Result<(), ()> {
   |  ________________________________________________________________^
29 | |     Ok(())?;
30 | |     Ok(());
   | |           - help: remove this semicolon to return this value
31 | | }
   | |_^ expected `Result<(), ()>`, found `()`
   |
   = note:   expected enum `Result<(), ()>`
           found unit type `()`
